Дз 4

SOProfile opened the pull request:
Работа со stream api

Commit:
добавлены тесты и классы исключений

Commit:
добавлены тесты и классы исключений

Commit:
добавлены методы для:
1. поиска среднего возраста животных
2. поиска животных, возраст которых больше 5 и цена выше средней
3. поиска 3 животных с самой низкой ценой, отсортированных в обратном алф. порядке

методы класса animalRepositoryImpl переведены на stream api

NosikovNeOleg commented on src/main/java/SearchServiceImpl.java:
```diff
@@ -14,9 +14,9 @@ public String checkLeapYearAnimal(Animal animal) throws InvalidAnimalBirthDateEx
 
         if ( Year.of(birthDate.getYear()).isLeap()) {
-            return "Год високосный";
+            return true;
```
Можно сразу вернуть результат работы метода isLeap

Commit:
добавлены методы для:
1. поиска среднего возраста животных
2. поиска животных, возраст которых больше 5 и цена выше средней
3. поиска 3 животных с самой низкой ценой, отсортированных в обратном алф. порядке
4. поправила метод для проверки года рождения животного на високосность

методы класса animalRepositoryImpl переведены на stream api
